tgui/packages/tgui/interfaces/Airlock.js

```diff
@@ -1,8 +1,417 @@
+              align="center"
+              mt={0.5}
+              color={opened ? "bad" : "good"}
+              icon={opened ? "sign-out-alt" : "sign-in-alt"}
+              disabled={(!boltsAreUp || welded)
+                || (mainTimeLeft && backupTimeLeft) || (isDisabled)}
+              onClick={() => act("openClose")}>
+              {opened ? "Open" : "Closed"}
+            </Button>
+          )}>
+          {!!(!boltsAreUp || welded) && (
+            <span>
+              [Door is {!boltsAreUp ? "bolted" : ""}
+              {(!boltsAreUp && welded) ? " and " : ""}
+              {welded ? "welded" : ""}!]
+            </span>
+          )}
+        </LabeledList.Item>
+      </LabeledList>
+    </Section>
+  );
+};
+
+
+const Electrify = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    mainTimeLeft,
+    backupTimeLeft,
+    wires,
+    shockTimeLeft,
+    netId,
+  } = data;
+
+  return (
+    <NoticeBox danger>
+      <Section m={-1} py={0.5}>
+        <LabeledList>
+          <LabeledList.Item
+            color={shockTimeLeft ? "average" : "good"}
+            label="Electrify">
+            {!shockTimeLeft ? "Safe" : "Electrified"}
+            {" "}
+            {!wires.shock
+            && "[Wires cut!]"
+            || (shockTimeLeft > 0
+            && `[${shockTimeLeft}s]`)
+            || (shockTimeLeft === -1
+            && "[Permanent]")}
+          </LabeledList.Item>
+          <LabeledList.Item
+            color={!shockTimeLeft ? "Average" : "Bad"}>
+            <Box
+              pl={shockTimeLeft ? 21 : 0}
+              pb={0}
+              pt={0.5}>
+              {(!shockTimeLeft &&(
+                <Button.Confirm
+                  p={1}
+                  width={9}
+                  align="center"
+                  color="average"
+                  content="Temporary"
+                  confirmContent="Are you sure?"
+                  icon="bolt"
+                  disabled={(!wires.shock) || shockTimeLeft === -1
+                || (mainTimeLeft && backupTimeLeft)}
+                  onClick={(() => act("shockTemp"))} />
+              ))}
+              <Button.Confirm
+                p={1}
+                width={9}
+                align="center"
+                color={shockTimeLeft ? "good" : "bad"}
+                icon="bolt"
+                confirmContent="Are you sure?"
+                content={shockTimeLeft ? "Restore" : "Permanent"}
+                disabled={(!wires.shock)
+                || (mainTimeLeft && backupTimeLeft)}
+                onClick={shockTimeLeft ? (() => act("shockRestore"))
+                  : (() => act("shockPerm"))} />
+            </Box>
+          </LabeledList.Item>
+        </LabeledList>
+      </Section>
+    </NoticeBox>
+  );
+};
+
+
+const Hack = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    aiHacking,
+  } = data;
+
+  return (
+    <Box
+      m={-1} py={0.5} pt={2}
```

Comment:
`-1` margin seems weird. Is this to make it so that it sits flush within the box/section above which normally has padding? Could you add the `fitted` prop to that container and apply appropriate margin to the other children? I would prefer applying margin a few times than applying negative margin once, as it's easier to understand.